olive/passes/qnn/conversion.py

```diff
@@ -79,38 +79,46 @@ def _run_for_config(
         else:
             # TODO(trajep): add tflite support
             raise NotImplementedError(f"Unsupported model handler type: {type(model)}")
-        converter_program = f"qnn-{converter_platform}-converter"
+        converter_program = [f"qnn-{converter_platform}-converter"]
 
         runner = QNNSDKRunner(use_dev_tools=True)
         if platform.system() == "Windows":
-            converter_program = "python " + str(
-                Path(runner.sdk_env.sdk_root_path) / "bin" / runner.sdk_env.target_arch / converter_program
-            )
+            converter_program = [
+                "python",
+                str(Path(runner.sdk_env.sdk_root_path) / "bin" / runner.sdk_env.target_arch / converter_program),
+            ]
 
         # get input dim from io_config
         input_dims = None
         if config.get("input_dim"):
             input_dims = config["input_dim"]
         elif model.io_config:
-            input_dims_tuple = zip(model.io_config.input_names, model.io_config.input_shapes)
-            # '{name}' is required to wrap the input name
-            input_dims = [f"'{name}' {','.joint(shape)}" for name, shape in input_dims_tuple]
+            input_dims_tuple = zip(model.io_config["input_names"], model.io_config["input_shapes"])
+            input_dims = [[name, ",".join(map(str, shape))] for name, shape in input_dims_tuple]
 
         out_nodes = None
         if config.get("out_node"):
             out_nodes = config["out_node"]
         elif model.io_config:
-            out_nodes = model.io_config.output_names
+            out_nodes = model.io_config["output_names"]
 
         output_model_path = normalize_path_suffix(output_model_path, "model.cpp")
-
+        # TODO(anyone): unify other cmd to list to avoid shlex.split error in non-posix mode
         cmd_list = [
-            converter_program,
-            f"--input_network {model.model_path}",
-            f"--output_path {output_model_path}",
-            " ".join([f"--input_dim {i}" for i in input_dims]) if input_dims else "",
-            " ".join([f"--out_node {o}" for o in out_nodes]) if out_nodes else "",
-            config["extra_args"] or "",
+            *converter_program,
+            "--input_network",
+            model.model_path,
+            "--output_path",
+            output_model_path,
         ]
-        runner.run(" ".join(cmd_list))
+        if input_dims:
+            for i in input_dims:
+                cmd_list.extend(["--input_dim", *i])
+        if out_nodes:
+            for o in out_nodes:
+                cmd_list.extend(["--out_node", o])
+        if config["extra_args"]:
+            cmd_list.extend(config["extra_args"].split())
+
+        runner.run(cmd_list)
```

Comment:
the runner doesn't support a list yet https://github.com/microsoft/Olive/blob/d068cf963e0ec1c5d4ecf9b87c2a1e118fe2303f/olive/platform_sdk/qualcomm/runner.py#L49.

we could convert str command to list around line 48 and then make everything else after it operate on the list command?

Reply:
I do agree with the TODO to make all commands list. Since the shlex with posix and not posix deals with "'str'" and paths differently. 

I also encountered similar issues before when using single quotes.

Accepting only lists takes away the responsibility from the subprocess runner to parse it.

Reply:
agree

Reply:
Oh, it is True. Let me upate the resolve part to let it accept list as well to return the value of the promise.

Reply:
should we avoid using shlex.split by favor string.split?

Reply:
Updated and keep the input to `_resolve_cmd` is string as of know. Also added the TODO: use list instead of string to avoid shlex.split error in non-posix mode

Reply:
do you mean line 68 here? https://github.com/microsoft/Olive/blob/d068cf963e0ec1c5d4ecf9b87c2a1e118fe2303f/olive/platform_sdk/qualcomm/runner.py#L68 

yes, I think that would be safer. Since we know the commands here of the type `command --arg ...`, string split will work. 
shlex.split is only useful if we are using some complex bash scripts that need the args to split correctly.

Reply:
do you mean line 68 here? https://github.com/microsoft/Olive/blob/d068cf963e0ec1c5d4ecf9b87c2a1e118fe2303f/olive/platform_sdk/qualcomm/runner.py#L68 

yes, I think that would be safer. Since we know the commands here of the type `command --arg ...`, string split will work. 
shlex.split is only useful if we are using some complex bash scripts that need the args to split correctly.

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/841ed3a9-4cac-4c91-a219-e1a906c74464)
The qnn/conversion. Just ensure our ci can pass and try to use List in separated PR.

Reply:
just change the _resolve_cmd to support both string and list